seqr/utils/elasticsearch/es_search.py

```diff
@@ -206,11 +208,15 @@ def filter_by_annotation_and_genotype(self, inheritance, quality_filter=None, an
             for index in self._indices:
                 family_samples_by_id = self.samples_by_family_index[index]
                 affected_status = _get_family_affected_status(family_samples_by_id, inheritance_filter)
-                self._family_individual_affected_status.update(affected_status)
+                for family_guid in affected_status:
+                    if family_guid in self._family_individual_affected_status:
+                        self._family_individual_affected_status[family_guid].update(affected_status[family_guid])
+                    else:
+                        self._family_individual_affected_status[family_guid] = affected_status[family_guid]
```

Comment:
I'm not sure it is correct or not. The original code replaces the `affected_status` of a family with `family_guid` rather than updates them. I have changed it to update the `affected_status` of the family if the status exists.

Reply:
This probably does the same thing but possibly doesn't and is way longer and less readable so why change it?

Reply:
If we agree on sorting `self._indices`, we can keep the existing code and it has the same behavior in both Python 2 and 3. If we don't sort `self._indices`, the result of the code will be different for Python 2 and 3. The new code doesn't depend on the order of `self._indices`.
I can change it back for now.